### ISSUE TITLE
False-positive for PRMC_POSSIBLY_REDUNDANT_METHOD_CALLS

### ISSUE THREAD

**Issue by Vampire** [spotbugs/spotbugs#966]
Following code triggers a false-positive for `PRMC_POSSIBLY_REDUNDANT_METHOD_CALLS` for the `length` call:
```java
left.insert(left.length() - 2, "(?:");
left.insert(left.length() - 2, PARAMETER_SEPARATOR_PATTERN_STRING);
left.insert(left.length() - 2, "|$)");
```
It seems the check misses that this is used as parameter to an `insert` call on that same instance which then of course is evaluated after the `length()` call and thus changes the returned value.

**Comment by ThrawnCA** [comment]
What is `left`, a StringBuilder?

**Comment by Vampire** [comment]
Ah, sorry, yes it is

**Comment by YinZho** [comment]
what is `PARAMETER_SEPARATOR_PATTERN_STRING` ?

**Comment by Vampire** [comment]
A string

**Comment by Xiaoven** [comment]
It seems PRMC_POSSIBLY_REDUNDANT_METHOD_CALLS is a pattern of [fb-contrib](http://fb-contrib.sourceforge.net/bugdescriptions.html#PRMC_POSSIBLY_REDUNDANT_METHOD_CALLS), not  of spotBugs.

**Comment by Vampire** [comment]
Oh, my bad, sorry.
I wish someone had said this two years ago. :-(
https://github.com/mebigfatguy/fb-contrib/issues/416
